src/Compilers/Server/ServerShared/BuildServerController.cs

```diff
@@ -29,7 +29,9 @@ internal int Run(string[] args)
             }
 
             pipeName = pipeName ?? GetDefaultPipeName();
+#pragma warning disable CA2000 // Dispose objects before losing scope - cancellationTokenSource escaped with the capture in CancelKeyPress event.
             var cancellationTokenSource = new CancellationTokenSource();
+#pragma warning restore CA2000 // Dispose objects before losing scope
             Console.CancelKeyPress += (sender, e) => { cancellationTokenSource.Cancel(); };
```

Comment:
Why are we suppressing the warning here vs. fixing the problem? 

---
In reply to: [171943963](https://github.com/dotnet/roslyn/pull/25179#discussion_r171943963) [](ancestors = 171943963)

Reply:
After chatting with @jinujoseph, we have decided to fix just absolutely trivial, non-contentious issues as part of this PR, so that we do not increase the scope of this PR anymore. I will file a follow up issue to deal with all the suppressions for dispose violations.

Reply:
Fixing this is no more complicated than any of the other fixes in this PR. The use of `cancellationTokenSource` should be wrapped in a `using` that covers the `return` statement. 

---
In reply to: [178571434](https://github.com/dotnet/roslyn/pull/25179#discussion_r178571434) [](ancestors = 178571434)

Reply:
> The use of cancellationTokenSource should be wrapped in a using that covers the return statement.

We cannot dispose the cancellation source until we unregister the event hooked up to `Console.CancelKeyPress`.

> Fixing this is no more complicated than any of the other fixes in this PR.

That was my initial attempt with this PR, but it has increased the scope of the PR significantly. I am going to revert majority of the fixes from my previous commits so they can be addressed/discussed with a separate issue and PR. The goal for this PR is to move us to newer analyzer packages and I would like to get it in soon to unblock @DustinCampbell.